preload.js

```diff
@@ -189,6 +191,17 @@ window.getSettingValue = (settingID, comparisonValue = null) => {
   // Comparison value allows you to pull boolean values from any type.
   // Eg. window.getSettingValue('theme', 'light')
   // returns 'false' when the value is 'dark'.
+
+  if (settingID === 'media-permissions') {
+    let permissionValue;
+    // eslint-disable-next-line more/no-then
+    window.getMediaPermissions().then(value => {
+      permissionValue = value;
+    });
+
+    return permissionValue;
+  }
```

Comment:
That does not work. 
you need to `await` `getMediaPermissions()` here rather than `.then()` it.

Doing it your way,
You are firing `window.getMediaPermissions` in async, then returning `permisssionValue` with its default value.

Later, your `then()` is called and does nothing.

Reply:
Thanks

Reply:
Solved